### PR TITLE
AP-3048 Convert documentation to Swagger

### PR DESCRIPTION
**Pull request by stephenrichards**
## Convert documentation to Swagger

Previously, documentaiton was generated using APIPIE.  This change converts the service
to using swagger documentation generated by RSWAG for the followiong reasons:
 - APIPIE is no longer maintained
 - APIPIE needs jQuery which we would prefer not to use
 - Swagger documentation is more comprehensive and generates a page where payloads can be tested on
   real servers

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3048)

 - ripped out all APIPIE DSL in controllers
 - removed APIPIE gem and replaced with rswag gem
 - removed APIPIE validators, initializers and layouts
 - Wrote new specs with rswag DSL to generate documentation in `spec/requests/swagger_docs`
 - Wrote rake task `rake swag` to easily generate new docs
 - Updated README
 - Added RSpec filter config rule to exclude swagger specs from normal run

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -40,6 +40,7 @@ group :development, :test do
   gem "pry-byebug"
   gem "rspec_junit_formatter"
   gem "rspec-rails", "~> 5.1"
+  gem "rswag"
```
Comment (willc-work) [inline Gemfile:43]: this needs to move to outside the dev/test section as we need it in production for UAT/STG/PROD deployments

Comment (willc-work) [reply Gemfile:43]: also the gem "apipie-rails", ">= 0.7.1" [line 31] can be removed from gemfile

File: spec/rails_helper.rb [hunk 2]
```diff
@@ -64,6 +64,9 @@
   config.filter_rails_from_backtrace!
   # arbitrary gems may also be filtered via:
   # config.filter_gems_from_backtrace("gem name")
+
+  # This shuld exclude all specs with swagger: true metadata
```
Comment (chris-groves) [inline spec/rails_helper.rb:68]: small typo in the spelling of 'should'

File: spec/services/proceeding_type_populator_spec.rb [hunk 3]
```diff
@@ -14,7 +16,7 @@
     end
 
     context "when run twice" do
-      it "creates the same total number of proceeding_types" do
+      it "creates §the same total number of proceeding_types" do
```
Comment (chris-groves) [inline spec/services/proceeding_type_populator_spec.rb:19]: can the § be removed?

**Comment by chris-groves** [comment]
some of the example data at http://localhost:3000/api-docs/index.html has empty arrays. Also have another look at the example response for a bad request to the GET proceeding types endpoint.

**Pull request merged**